petri/logview_new/src/log_viewer.tsx

```diff
@@ -0,0 +1,432 @@
+        if (selection && !selection.isCollapsed && selection.toString().trim().length > 0) return;
+
+        // Need a selected log row
+        if (!selectedRow) return;
+
+        const idx = parseInt(selectedRow.replace('log-', ''), 10);
+        if (isNaN(idx)) return;
+        const entry = logEntries?.find(le => le.index === idx);
+        if (!entry) return;
+
+        if (e.shiftKey) {
+            // Ctrl+Shift+C (or Cmd+Shift+C) => copy deep link to this log line (hash-based routing aware)
+            e.preventDefault();
+            const { origin, pathname, hash } = window.location;
+            // HashRouter format: <origin><pathname>#/route/segments?query
+            const hashParts = hash.split('?');
+            const hashRoute = hashParts[0] || '#/';
+            const hashQuery = hashParts[1] ? new URLSearchParams(hashParts[1]) : new URLSearchParams();
+            hashQuery.set('log', String(entry.index));
+            const deepLink = `${origin}${pathname}${hashRoute}?${hashQuery.toString()}`;
+            navigator.clipboard?.writeText(deepLink).catch(() => {
+                try {
+                    const ta = document.createElement('textarea');
+                    ta.value = deepLink; ta.style.position = 'fixed'; ta.style.opacity = '0';
+                    document.body.appendChild(ta); ta.select(); document.execCommand('copy'); document.body.removeChild(ta);
+                } catch { /* no-op */ }
+            });
+            return;
+        }
+
+        // Plain Ctrl+C => copy plain text representation of the selected log line
+        e.preventDefault();
+        
+        // Decode HTML entities in the message text
+        const decodeHtml = (html: string): string => {
+            const txt = document.createElement('textarea');
+            txt.innerHTML = html;
+            return txt.value;
+        };
+        
+        let textBlock = `timestamp: ${entry.timestamp}\n`;
+        textBlock += `relative: ${entry.relative}\n`;
+        textBlock += `severity: ${entry.severity}\n`;
+        textBlock += `source: ${decodeHtml(entry.source)}\n`;
+        textBlock += `message: ${decodeHtml(entry.messageText.trim())}`;
+        if (entry.screenshot) {
+            textBlock += `\nscreenshot: ${entry.screenshot}`;
+        }
+        
+        navigator.clipboard?.writeText(textBlock).catch(() => {
+            try {
+                const ta = document.createElement('textarea');
+                ta.value = textBlock; ta.style.position = 'fixed'; ta.style.opacity = '0';
+                document.body.appendChild(ta); ta.select(); document.execCommand('copy'); document.body.removeChild(ta);
+            } catch { /* no-op */ }
+        });
+    };
+}
+
+/**
+ * Filters log entries based on a search query.
+ * 
+ * The query is tokenized by whitespace, with support for quoted phrases.
+ * Multiple tokens act as AND filters - all tokens must match for the log to pass.
+ * 
+ * Special field-specific search syntax:
```

Comment:
nice! can you just make this an example in the text box? thinking about how folks would discover this. (this is a feature request, not a request for changes on this PR!)